castervoice/lib/ccr/core/nav.py

```diff
@@ -268,15 +270,68 @@ class Navigation(MergeRule):
+        "bird [<nnavi500>]": R(Key("c-left:%(nnavi500)s")),
+        "fird [<nnavi500>]": R(Key("c-right:%(nnavi500)s")),
+        "brick [<nnavi500>]": R(Key("s-left:%(nnavi500)s")),
+        "frick [<nnavi500>]": R(Key("s-right:%(nnavi500)s")),
+        "blitch [<nnavi500>]": R(Key("cs-left:%(nnavi500)s")),
+        "flitch [<nnavi500>]": R(Key("cs-right:%(nnavi500)s")),
+        
+        "<modifier> <button_dictionary_500> [<nnavi500>]":
+              R(Key("%(modifier)s-%(button_dictionary_500)s") * Repeat(extra='nnavi500'), 
+              rdescript="press modifier keys plus buttons from button_dictionary_500"),
+        "<modifier> <button_dictionary_10> [<nnavi10>]":
+              R(Key("%(modifier)s-%(button_dictionary_10)s") * Repeat(extra='nnavi10'),
+              rdescript="press modifier keys plus buttons from button_dictionary_10"),
+        "<modifier> <button_dictionary_1>":
+              R(Key("%(modifier)s-%(button_dictionary_1)s"),
+              rdescript="press modifiers plus buttons from button_dictionary_1, non-repeatable"),
+        
+        
     }
+    # I tried to limit which things get repeated how many times in hopes that it will help prevent the bad grammar error
+    # this could definitely be changed. perhaps some of these should be made non-CCR
+    button_dictionary_500 = {"(tab | tabby)": "tab", "backspace": "backspace", "delete": "del", "(escape | cancel)": "escape", "(enter | shock)": "enter",
+    "(left | lease)": "left", "(right | ross)": "right", "(up | sauce)": "up",
+    "(down | dunce)": "down", "page (down | dunce)": "pgdown", "page (up | sauce)": "pgup", "space": "space"}
+    button_dictionary_10 = {}
+    button_dictionary_10.update(caster_alphabet)
+    button_dictionary_10.update(text_punc_dict)
+    button_dictionary_1 = {"home": "(home | lease wally | latch)", "(end | ross wally | ratch)": "end", "insert": "insert", "zero": "0",
+    "one": "1", "two": "2", "three": "3", "four": "4", "five": "5", "six":"6", "seven": "7", "eight": "8", "nine": "9"}
 
     extras = [
+        
         IntegerRefST("nnavi10", 1, 11),
         IntegerRefST("nnavi50", 1, 50),
         IntegerRefST("nnavi500", 1, 500),
         Dictation("textnv"),
         Choice("enclosure", double_text_punc_dict),
+        Choice("direction", {
+            "dunce": "down",
+            "sauce": "up",
+            "lease": "left",
+            "ross": "right",
+        }),
+        
+        Choice("button_dictionary_1", button_dictionary_1),
+        Choice("button_dictionary_10", button_dictionary_10), 
+        Choice("button_dictionary_500", button_dictionary_500), 
+        Choice("modifier", {
+            "(control | fly)": "c",
+            "(shift | shin)": "s",
+            "alt": "a",
+            "(control shift | que)": "cs",
+            "control alt": "ca",
+            "(shift alt | alt shift)": "sa",
+            "(control alt shift | control shift alt)": "csa", # control must go first
```

Comment:
allow composition with fly, shin, alt as well. ie "fly alt shin" should be an option

Reply:
Add "cass" as an option?

Reply:
The shortened names for the rarer modifiers

Reply:
Don't feel strongly about it either way. People can add them